src/protocol/templates/pingmessage.h.in

```diff
@@ -26,49 +26,88 @@ namespace {{capitalize(message_type)}}Namespace {
 };
 {% endfor %}
 
-class {{class_info.name}} // TODO inherit vector?
+class {{class_info.name}}
 {
 public:
-    {{class_info.name}}() {}
-    {{class_info.name}}(QByteArray& buf) { msgData = std::vector<uint8_t>(buf.begin(), buf.end()); }
-    {{class_info.name}}(std::vector<uint8_t>& buf) { msgData = buf; }
-    {{class_info.name}}({{class_info.name}}& msg) { msgData = msg.msgData; }
-    ~{{class_info.name}}() {}
+    {{class_info.name}}() : msgData(nullptr) {}
 
-    std::vector<uint8_t> msgData;
+    {{class_info.name}}(const {{class_info.name}} &msg)
+        : _bufferLength(msg.msgDataLength())
+    {
+        msgData = (uint8_t*)malloc(sizeof(uint8_t) * _bufferLength);
+        memcpy(msgData, msg.msgData, _bufferLength);
+    }
+
+    PingMessage(uint16_t bufferLength)
+        : _bufferLength(bufferLength)
+    {
+        msgData = (uint8_t*)malloc(sizeof(uint8_t) * _bufferLength);
```

Comment:
can this be done in the constructor initialization list ?
```cpp
        : _bufferLength(bufferLength)
        , msgData{(uint8_t*)malloc(sizeof(uint8_t) * _bufferLength)};
    {
```
I know that can be done with new.
```
        : _bufferLength(bufferLength)
        , msgData{new uint8_t[_bufferLength]};
    {
```